pkg/common/cns-lib/volume/manager.go

```diff
@@ -565,63 +576,184 @@ func (m *defaultManager) DeleteVolume(ctx context.Context, volumeID string, dele
+			volumeOperationDetails = createRequestDetails(instanceName, "", "", 0,
+				metav1.Now(), "", "", taskInvocationStatusError, err.Error())
 			return err
 		}
-		// Get the taskInfo.
-		taskInfo, err := cns.GetTaskInfo(ctx, task)
-		if err != nil || taskInfo == nil {
-			log.Errorf("failed to get DeleteVolume taskInfo from vCenter %q with err: %v",
-				m.virtualCenter.Config.Host, err)
-			return err
-		}
-		log.Infof("DeleteVolume: volumeID: %q, opId: %q", volumeID, taskInfo.ActivationId)
-		// Get the task results for the given task.
-		taskResult, err := cns.GetTaskResult(ctx, taskInfo)
+		volumeOperationDetails = createRequestDetails(instanceName, "", "", 0, metav1.Now(),
+			task.Reference().Value, "", taskInvocationStatusInProgress, "")
+		err := m.operationStore.StoreRequestDetails(ctx, volumeOperationDetails)
 		if err != nil {
-			log.Errorf("unable to find DeleteVolume task result from vCenter %q with taskID %s and deleteResults %v",
-				m.virtualCenter.Config.Host, taskInfo.Task.Value, taskResult)
-			return err
-		}
-		if taskResult == nil {
-			log.Errorf("taskResult is empty for DeleteVolume task: %q, opID: %q",
-				taskInfo.Task.Value, taskInfo.ActivationId)
-			return errors.New("taskResult is empty")
-		}
-		volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
-		if volumeOperationRes.Fault != nil {
-			msg := fmt.Sprintf("failed to delete volume: %q, fault: %q, opID: %q",
-				volumeID, spew.Sdump(volumeOperationRes.Fault), taskInfo.ActivationId)
-			log.Error(msg)
-			return errors.New(msg)
+			log.Warnf("failed to store DeleteVolume details with error: %v", err)
 		}
-		log.Infof("DeleteVolume: Volume deleted successfully. volumeID: %q, opId: %q",
-			volumeID, taskInfo.ActivationId)
-		return nil
 	}
-	start := time.Now()
-	err := internalDeleteVolume()
+
+	// Get the taskInfo.
+	taskInfo, err := cns.GetTaskInfo(ctx, task)
+	if err != nil || taskInfo == nil {
+		log.Errorf("failed to get taskInfo for DeleteVolume task from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
+		return err
+	}
+
+	log.Infof("DeleteVolume: volumeID: %q, opId: %q", volumeID, taskInfo.ActivationId)
+	// Get the task results for the given task.
+	taskResult, err := getTaskResultFromTaskInfo(ctx, taskInfo)
 	if err != nil {
-		prometheus.CnsControlOpsHistVec.WithLabelValues(prometheus.PrometheusCnsDeleteVolumeOpType,
-			prometheus.PrometheusFailStatus).Observe(time.Since(start).Seconds())
-	} else {
-		prometheus.CnsControlOpsHistVec.WithLabelValues(prometheus.PrometheusCnsDeleteVolumeOpType,
-			prometheus.PrometheusPassStatus).Observe(time.Since(start).Seconds())
+		log.Errorf("unable to find DeleteVolume task result from vCenter %q with taskID %s and deleteResults %v",
+			m.virtualCenter.Config.Host, taskInfo.Task.Value, taskResult)
+		return err
 	}
-	return err
+
+	volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
+	if volumeOperationRes.Fault != nil {
```

Comment:
Can we check if we will get NotFound fault here?

Reply:
I've reached out to CNS about this. Since this is the existing behavior i wont change it for now

Reply:
Tracking the work in #949
Assigned it to myself